recipe/meta.yaml

```diff
@@ -1,34 +1,42 @@
-# Using the name variable with the URL in line 13 is conviniet
-# when copying and pasting from another recipe, but not really needed.
-{% set name = "fluidsynth" %}
 {% set version = "2.2.8" %}
 
 package:
-  name: {{ name|lower }}
+  name: fluidsynth
   version: {{ version }}
 
 source:
-  url: https://github.com/FluidSynth/{{ name }}/archive/v{{ version }}.tar.gz
+  url: https://github.com/FluidSynth/fluidsynth/archive/v{{ version }}.tar.gz
   sha256: 7c29a5cb7a2755c8012d941d1335da7bda957bbb0a86b7c59215d26773bb51fe
+  patches:
+    # upstream uses a hack to ensure an executable built for the
+    # build process itself has the right (=host) CPU type;
+    # the hack unsets the CC variable, meaning it'll pick up the host
+    # compiler (on osx: xcode) rather than our own infrastructure;
+    # obviously we should do this properly also on osx-arm
+    - patches/0001-don-t-unset-required-environment-flags.patch  # [not (osx and arm64)]
```

Comment:
@isuruf 

JFYI - upstream there's a package-in-package build that needs to build a (custom) binary for the build process, obviously with the host arch. Upstream did this by unsetting `$CC` and a bunch of other things, and 🤞 that the right host compiler gets picked.

For now I've left this patch in effect for osx-arm, but this is clearly not what we want to do, but I'm not sure how to pass the right arguments to that nested CMake build so it picks up our cmake infra, but compiles for host even when all other CMAKE_* options are geared for cross-compilation.

The good thing is that the impact of this is minimal, because it's just a generator that doesn't packaged. Still if you know how to solve this properly, I'd be happy to hear it.